esphomeyaml/cookbook/sonoff-fishpond-pump.rst

```diff
@@ -0,0 +1,367 @@
+
+Assumptions
+-----------
+
+As many of the details on here a predicated on existing installations and knowledge the following is assumed:
+
+* You have installed the relevant environment and it is functional (Home Assisant, esphomeyaml).
+* You need to create your own secrets.yaml file.
+* You are proficient in reading instructions and capable of using a search engine.
+* You will look at the documentation on https://esphomelib.com/ (we may have missed something in this example).
+* This serves only as a guide and errors and outlets are excluded.
+
+.. warning::
+
+    Do NOT connect your device to electricity when programming it.
+    Take care working with electricity at all times
+
+
+1. Configuration
+----------------
+
+Here is the configuration with the basic operations outlined above.
+
+.. code-block:: yaml
+
+    ## Fish pond sonoff
+    ## filename: fishpond.yaml
+    # Sonoff basic
+    # Starts with pump in on position for 5 minutes
+    # Button toggles pump on/off
+    # Float sensor stops pump
+    # Cron'd to run every hour, on the hour for 10 minutes
+    esphomeyaml:
+      name: !secret fishpond
+      platform: ESP8266
+      board: esp01_1m
+      board_flash_mode: dout
+
+      # run on boot for 5 minutes
+      on_boot:
+        priority: 50
+        then:
+          - switch.turn_on: esp_fishpond_pump
+          - delay: 5min
+          - switch.turn_off: esp_fishpond_pump
+
+    wifi:
+      ssid: !secret wifi_ssid
+      password: !secret wifi_password
+      domain: !secret wifi_domain
+      fast_connect: true
+
+    # Enable logging
+    logger:
+      level: DEBUG
+
+    ota:
+
+    api:
+      password: !secret esplibapi_password
+
+    status_led:
+      pin: GPIO13
+
+    sensor:
+      - platform: uptime
+        name: "ESP Fish Pond System Uptime"
+
+      - platform: wifi_signal
+        name: "ESP Fish Pond WiFi Signal"
+        update_interval: 15s
+
+    switch:
+      ## restart
+      - platform: restart
+        name: "ESP Fishpond Restart"
+
+      ## relay / pump
+      - platform: gpio
+        pin: GPIO12
+        name: "ESP Fishpond Pump"
+        id: esp_fishpond_pump
+
+    binary_sensor:
+      - platform: status
+        name: "ESP Fish Pond System Status"
+
+      ## button - toggle relay
+      - platform: gpio
+        name: "ESP Fish Pond Button"
+        pin:
+          number: GPIO0
+          inverted: true
+        # Toggle pump when user presses the Sonoff Button
+        on_press:
+          - switch.toggle: esp_fishpond_pump
+
+      # onboard header pin - sensor
+      # when threshold is reached, turn off
+      - platform: gpio
+        name: "esp_fishpond_gpio14"
+        pin:
+          number: 14
+          mode: INPUT_PULLUP
+        on_press:
+          - switch.turn_off: esp_fishpond_pump
+
+    ## Use time
+    time:
+      - platform: homeassistant
+        timezone: Africa/Johannesburg
+        on_time:
+          # Switch on on the hour
+          - seconds: 0
```

Comment:
I noticed in the debug logs running mqtt (v1.8.ish) once the first minute starts it sends the 'ON' signal constantly for the first minute ~60 commands to go 'ON'.

This became an issue when the cron kicks off and you need to stop / OFF the device with the toggle button, it just goes back on.

After an indecent with the running cron and toggle button not actually stopping the device it have added it in as a safety measure.

I will highly request this to be present and not removed. I have commented it as well in the config.

Reply:
Yes if you don't specify `seconds` it will default to `*`, which will make it trigger for every second in the first minute of each hour. So as a pattern it would look like this: `**:00:**`.

0-5 indeed does solve the problem, but using 0-5 this trigger will execute 6 times. For example at 13:00 it will execute at 13:00:00, 13:00:01, 13:00:02, ..., 13:00:05

If `0-5` is replaced with `0` it will only trigger once, at exactly `13:00:00`

Reply:
Ok, in that case I will change accordingly (will also test personally)